src/queryRewriters/rewriters/patientProxyQueryRewriter.js

```diff
@@ -28,6 +27,55 @@ class PatientProxyQueryRewriter extends QueryRewriter {
         assertTypeEquals(personToPatientIdsExpander, PersonToPatientIdsExpander);
     }
 
+    /**
+     * updates the queryParameters
+     * @param {ParsedArgs} parsedArgs
+     * @param {string} base_version
+     * @param {boolean} includePatientPrefix
+     * @returns {ParsedArgsItem}
+     */
+    async rewriteQueryParametersAsync({ parsedArg, base_version, includePatientPrefix }) {
+        const queryParameterValues = parsedArg.queryParameterValue.values;
+        if (queryParameterValues && queryParameterValues.length > 0) {
+            /**
+             * @type {string[]}
+             */
+            const queryParameterValuesToProcess = queryParameterValues.filter(a =>
```

Comment:
use `.reduce` function to get `queryParameterValuesToProcess` and `queryParameterValuesToNotProcess`  instead of using `.filter` two times.